pallets/proofs-dealer/src/benchmarking.rs

```diff
@@ -45,9 +83,192 @@ mod benchmarks {
+                    <T as pallet_storage_providers::Config>::StartingReputationWeight::get(),
+                sign_up_block: Default::default(),
+            },
+        );
+
+        // Hold some of the Provider's balance so it simulates it having a stake.
+        assert_ok!(<T as crate::Config>::NativeBalance::hold(
+            &pallet_storage_providers::HoldReason::StorageProviderDeposit.into(),
+            &caller,
+            provider_balance / 100u32.into(),
+        ));
+
+        // Set Provider's root to be the one that matches the proofs that will be submitted.
+        let encoded_root = get_root();
+        let root = <T as frame_system::Config>::Hash::decode(&mut encoded_root.as_ref())
+            .expect("Root should be decodable as it is a hash");
+        pallet_storage_providers::BackupStorageProviders::<T>::mutate(&provider_id, |provider| {
+            provider.as_mut().expect("Provider should exist").root = root;
+        });
+
+        // Set Provider's last submitted proof block.
+        let current_tick = ChallengesTicker::<T>::get();
+        let last_tick_provider_submitted_proof = current_tick;
+        LastTickProviderSubmittedAProofFor::<T>::insert(
+            &provider_id,
+            last_tick_provider_submitted_proof,
+        );
+
+        // Set Provider's deadline for submitting a proof.
+        // It is the sum of this Provider's challenge period and the `ChallengesTicksTolerance`.
+        let providers_stake =
+            <ProvidersPalletFor<T> as ReadChallengeableProvidersInterface>::get_stake(provider_id)
+                .unwrap();
+        let challenge_period = crate::Pallet::<T>::stake_to_challenge_period(providers_stake);
+        let challenge_ticks_tolerance: BlockNumberFor<T> = ChallengeTicksToleranceFor::<T>::get();
+        let challenge_period_plus_tolerance = challenge_period + challenge_ticks_tolerance;
+        let prev_deadline = current_tick + challenge_period_plus_tolerance;
+        TickToProvidersDeadlines::<T>::insert(prev_deadline, provider_id, ());
+
+        // Advance to the next challenge the Provider should listen to.
+        let providers_stake =
+            <ProvidersPalletFor<T> as ReadChallengeableProvidersInterface>::get_stake(provider_id)
+                .unwrap();
+        let challenge_period = crate::Pallet::<T>::stake_to_challenge_period(providers_stake);
+        let current_block = frame_system::Pallet::<T>::block_number();
+        let challenge_block = current_block + challenge_period;
+        frame_system::Pallet::<T>::set_block_number(challenge_block);
+        // Advance less than `ChallengeTicksTolerance` blocks.
+        let challenge_ticks_tolerance: BlockNumberFor<T> = ChallengeTicksToleranceFor::<T>::get();
+        let current_block = frame_system::Pallet::<T>::block_number();
+        frame_system::Pallet::<T>::set_block_number(
+            current_block + challenge_ticks_tolerance - 1u32.into(),
+        );
```

Comment:
is it possible to have common utility functions that calculate these deadlines and challenges if it can be shared with the extrinsic implementations as well? if there is something custom to this benchmark impl, then forget it.

Reply:
Honestly, it's some code used in many tests' setup. It could be useful to have a utility function for the test setup, but then that wouldn't be included here necessarily 🤷🏻‍♂️